runtime/autoload/provider/clipboard.vim

```diff
@@ -55,13 +55,21 @@ endif
 let s:clipboard = {}
 
 function! s:clipboard.get(reg)
-  if s:selections[a:reg].owner > 0
-    return s:selections[a:reg].data
+  let reg = a:reg == '"' ? '+' : a:reg
+  if s:selections[reg].owner > 0
+    return s:selections[reg].data
   end
-  return s:try_cmd(s:paste[a:reg])
+  return s:try_cmd(s:paste[reg])
 endfunction
 
 function! s:clipboard.set(lines, regtype, reg)
+  if a:reg == '"'
+    call s:clipboard.set(a:lines,a:regtype,'+')
+    if s:copy['*'] != s:copy['+']
+      call s:clipboard.set(a:lines,a:regtype,'*')
+    end
+    return 0
```

Comment:
shouldn't we return the result of the recursed call? Also I just noticed that this function always returns 0. I think you explained it before, can you remind me why?

Also would be good to have an explicit `return 0` at the end of the function.